include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -1454,6 +1454,13 @@ class FEPointEvaluation
     const StridedArrayView<ScalarNumber, stride_view> &solution_values,
     const EvaluationFlags::EvaluationFlags            &integration_flags,
     const bool                                         sum_into_values);
+
+  /**
+   * Internal function to initialize the pointers of this class when an
+   * external MappingInfo has already queried the mapping and cell vertices.
+   */
+  void
+  reinit_from_external_mapping_info();
```

Comment:
```suggestion
  internal_reinit_single_cell_state_mapping_info();
```
This name would be more fitting. It calls `do_reinit()` with `numbers::invalid_unsigned_int` for `current_cell_index` and `current_face_number`, which means it assumes the `MappingInfo` object to be in `State::single_cell`.